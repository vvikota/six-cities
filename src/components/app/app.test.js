import React from 'react';
import renderer from 'react-test-renderer';
import App from './app.jsx';
import {MemoryRouter} from "react-router-dom";

const cityOffersMock = [
  {
    "city": {
      "name": `Amsterdam`,
      "location": {
        "latitude": 52.37454,
        "longitude": 4.897976,
        "zoom": 13
      }
    },
    "previewImage": `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
    "images": [`https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/9.jpg", "https://es31-server.appspot.com/six-cities/static/hotel/11.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/3.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/16.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`],
    "title": `Amazing and Extremely Central Flat`,
    "isFavorite": false,
    "isPremium": false,
    "rating": 3.2,
    "type": `room`,
    "bedrooms": 1,
    "maxAdults": 2,
    "price": 250,
    "goods": [`Breakfast`, `Laptop friendly workspace`, `Towels`, `Fridge`, `Air conditioning`, `Washer`, `Dishwasher`, `Baby seat`],
    "host": {
      "id": 25,
      "name": `Angelina`,
      "isPro": true,
      "avatarUrl": `img/avatar-angelina.jpg`
    },
    "description": `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    "location": {
      "latitude": 52.37154,
      "longitude": 4.889976,
      "zoom": 16
    },
    "id": 1
  }, {
    "city": {
      "name": `Amsterdam`,
      "location": {
        "latitude": 52.37454,
        "longitude": 4.897976,
        "zoom": 13
      }
    },
    "previewImage": `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
    "images": [`https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/9.jpg", "https://es31-server.appspot.com/six-cities/static/hotel/11.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/3.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/16.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`],
    "title": `Amazing and Extremely Central Flat`,
    "isFavorite": false,
    "isPremium": false,
    "rating": 3.2,
    "type": `room`,
    "bedrooms": 1,
    "maxAdults": 2,
    "price": 250,
    "goods": [`Breakfast`, `Laptop friendly workspace`, `Towels`, `Fridge`, `Air conditioning`, `Washer`, `Dishwasher`, `Baby seat`],
    "host": {
      "id": 25,
      "name": `Angelina`,
      "isPro": true,
      "avatarUrl": `img/avatar-angelina.jpg`
    },
    "description": `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    "location": {
      "latitude": 52.37154,
      "longitude": 4.889976,
      "zoom": 16
    },
    "id": 1
  }, {
    "city": {
      "name": `Amsterdam`,
      "location": {
        "latitude": 52.37454,
        "longitude": 4.897976,
        "zoom": 13
      }
    },
    "previewImage": `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
    "images": [`https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/9.jpg", "https://es31-server.appspot.com/six-cities/static/hotel/11.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/3.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/16.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`],
    "title": `Amazing and Extremely Central Flat`,
    "isFavorite": false,
    "isPremium": false,
    "rating": 3.2,
    "type": `room`,
    "bedrooms": 1,
    "maxAdults": 2,
    "price": 250,
    "goods": [`Breakfast`, `Laptop friendly workspace`, `Towels`, `Fridge`, `Air conditioning`, `Washer`, `Dishwasher`, `Baby seat`],
    "host": {
      "id": 25,
      "name": `Angelina`,
      "isPro": true,
      "avatarUrl": `img/avatar-angelina.jpg`
    },
    "description": `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    "location": {
      "latitude": 52.37154,
      "longitude": 4.889976,
      "zoom": 16
    },
    "id": 1
  }
];

const currentOfferMock = {
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.37454,
      "longitude": 4.897976,
      "zoom": 13
    }
  },
  "previewImage": `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
  "images": [`https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/9.jpg", "https://es31-server.appspot.com/six-cities/static/hotel/11.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/3.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/16.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`, `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`],
  "title": `Amazing and Extremely Central Flat`,
  "isFavorite": false,
  "isPremium": false,
  "rating": 3.2,
  "type": `room`,
  "bedrooms": 1,
  "maxAdults": 2,
  "price": 250,
  "goods": [`Breakfast`, `Laptop friendly workspace`, `Towels`, `Fridge`, `Air conditioning`, `Washer`, `Dishwasher`, `Baby seat`],
  "host": {
    "id": 25,
    "name": `Angelina`,
    "isPro": true,
    "avatarUrl": `img/avatar-angelina.jpg`
  },
  "description": `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
  "location": {
    "latitude": 52.37154,
    "longitude": 4.889976,
    "zoom": 16
  },
  "id": 1
};

const hotelCommentsMock = [
  {
    id: 1,
    user: {
      id: 1,
      isPro: true,
      name: `Alena`,
      avatarUrl: `url`,
    },
    rating: 10,
    comment: `text`,
    date: `april 2020`,
  }
];

const userInformationMock = {
  id: 10,
  email: `mail`,
  name: `Petr`,
  avatarUrl: `avater`,
  isPro: true,
};

it(`App is correcctly render`, ()=> {

  const tree = renderer.create(<MemoryRouter><App
    city={`Amstrdam`}
    changeCity={jest.fn()}
    cityList={[]}
    cityOffers={cityOffersMock}
    isAuthorizationRequired={true}
    sendAuthorizationRequest={jest.fn()}
    userInformation={userInformationMock}
    openDetailOffer={jest.fn()}
    currentOffer={currentOfferMock}
    hotelComments={hotelCommentsMock}
    nearOffers={cityOffersMock}
  /></MemoryRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
