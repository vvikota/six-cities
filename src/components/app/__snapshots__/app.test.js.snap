// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is correcctly render 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  vik792@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="cities tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <span>
                Amsterdam
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities__places-wrapper"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            0
             places to stay in Amsterdam
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  href="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom places__options"
            >
              <li
                className="places__option places__option--active"
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option"
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="cities__places-list places__list tabs__content"
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <div
              id="map"
              style={
                Object {
                  "height": "90vh",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>,
  ";",
]
`;
